src/services/db/GitHubService.ts

```diff
@@ -767,11 +755,13 @@ export default class GitHubService {
 
   async updateRepoState(
     sessionData: UserWithSiteSessionData,
-    { commitSha }: { commitSha: any }
+    { commitSha, branchName }: { commitSha: any; branchName?: string }
   ) {
     const { accessToken } = sessionData
     const { siteName } = sessionData
-    const refEndpoint = `${siteName}/git/refs/heads/${STAGING_BRANCH}`
+    const refEndpoint = `${siteName}/git/refs/heads/${
+      branchName || STAGING_BRANCH
+    }`
```

Comment:
i might be lacking context here but it seems abit weird to me to see `branchName` allowed as an argument here when we're explicitly removing it elsewhere (cos we know the branch should be `STAGING_BRANCH`) - is this because there are some call sites that need the branch to be different? (a merge to `master` perhaps)

Reply:
Yes, this is the case! Specifically we have a `RepoManagementService` that calls on this branch name, and thus this API should be preserved

Reply:
thanks for checking - seems like we (isomer admins) use it to reset branches so it should still be `staging`. that being said, it's ok to leave as-is cos this is hard to tell for sure + the added flexibility of resetting a diff branch might be useful for `staging/staging-lite/master`